src/app/common/filter-bar/FilterBar.vue

```diff
@@ -222,12 +222,15 @@ watch(() => currentFields.value, function (newFields, oldFields) {
   }
 })
 
+watch(() => props.query, () => {
+  currentQuery.value = props.query
+  recomputeFields(currentQuery.value)
+}, { immediate: true })
+
 watch(() => currentQuery.value, function () {
   if (currentQuery.value === '') {
     tokenizerError.value = null
   }
-
-  isShowingSuggestionBox.value = true
```

Comment:
How come these couple of changes were needed? (both the red and green in this file)

Reply:
With the changes here, what we now have are situations where the query parameter of a route can be programmatically changed through something that isn’t an already open filter box: clicking on a service tag in a dataplane summary view doesn’t navigate because the list view is already rendered, but it does populate the query in the URL. What happened then was that clicking on such a link would _open_ the suggestion box and that’s addressed with this change.

There’s also a change here that makes sure that changing `props.query` is reflected in `currentQuery`.

Reply:
Gotcha, thanks